clash-lib/src/Clash/Driver/Types.hs

```diff
@@ -401,6 +401,9 @@ data ClashOpts = ClashOpts
   , opt_timescalePrecision :: Period
   -- ^ Timescale precision set in Verilog files. E.g., setting this would sets
   -- the second part of @`timescale 100fs/100fs@.
+  , opt_sanitizeNames :: Bool
+  -- ^ Sanitize ID's for particular tools that may reject
+  -- otherwise valid names.
```

Comment:
Let's be clearer about this: we expect to sanitise some names anyway since this happens when you don't use `unsafeFromCoreId`. What this flag does is force names to be sanitised in places where they wouldn't normally be.

Reply:
Also done.